File: Microserivce_ECommerSystem/ReviewService/ReviewService/src/main/java/com/example/ReviewService/service/ReviewService.java
package com.example.ReviewService.service;

import com.example.ReviewService.dto.ReviewRequestDTO;
import com.example.ReviewService.dto.ReviewResponseDTO;

import java.util.List;

public interface ReviewService {
    ReviewResponseDTO createReview(ReviewRequestDTO request);
    List<ReviewResponseDTO> getReviewsByProduct(String productId);
    List<ReviewResponseDTO> getReviewsByUser(String userId);
    void deleteReview(String id);
}

File: FE/src/services/ReviewService.js
import axios from '../api/axios';

class ReviewService {
    async getReviewsByProduct(productId) {
        const res = await axios.get(`/api/reviews/product/${productId}`);
        return res.data;
    }

    async createReview(reviewData) {
        const res = await axios.post('/api/reviews', reviewData);
        return res.data;
    }

    async deleteReview(id) {
        const res = await axios.delete(`/api/reviews/${id}`);
        return res.data;
    }

    async getReviewsByUser(userId) {
        const res = await axios.get(`/api/reviews/user/${userId}`);
        return res.data;
    }
}

export default new ReviewService();
